### PR TITLE
Auth actions user input validation

### PR DESCRIPTION
**Pull request by Gligor23**
# Pull Request Template

## Description

Auth endpoints user input validation

## Type of change

Please delete options that are not relevant.

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A - test requests in insomnia
- [ ] Test B - integration tests

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: actions/auth.go [hunk 1]
```diff
@@ -32,13 +26,34 @@ import (
...
+	"github.com/mailbadger/app/utils"
+	"github.com/mailbadger/app/validator"
 )
 
 // PostAuthenticate authenticates a user with the given username and password.
 func PostAuthenticate(c *gin.Context) {
-	username, password := c.PostForm("username"), c.PostForm("password")
+	body := &params.PostAuthenticate{}
+	if err:=c.ShouldBind(body); err != nil {
+		logger.From(c).WithError(err).Error("Unable to bind authentication params.")
```
Comment (Dzalevski) [inline actions/auth.go:43]: @Gligor23 i think we don't need log for this. also on validate.

File: actions/auth.go [hunk 2]
```diff
@@ -32,13 +26,34 @@ import (
...
+	"github.com/mailbadger/app/storage"
+	"github.com/mailbadger/app/utils"
+	"github.com/mailbadger/app/validator"
 )
 
 // PostAuthenticate authenticates a user with the given username and password.
 func PostAuthenticate(c *gin.Context) {
-	username, password := c.PostForm("username"), c.PostForm("password")
+	body := &params.PostAuthenticate{}
+	if err:=c.ShouldBind(body); err != nil {
```
Comment (FilipNikolovski) [inline actions/auth.go:42]: Run gofmt.

File: actions/auth.go (hunk 1 above)
Comment (FilipNikolovski) [reply actions/auth.go:43]: Maybe we should have a lower level log, or is this something not worth logging?

File: actions/auth.go [hunk 3]
```diff
@@ -32,13 +26,34 @@ import (
...
+		logger.From(c).WithError(err).Error("Unable to bind authentication params.")
+		c.JSON(http.StatusBadRequest, gin.H{
+			"message": "Invalid parameters, please try again",
+		})
+		return
+	}
 
-	user, err := storage.GetActiveUserByUsername(c, username)
+	if err := validator.Validate(body); err != nil {
+		logger.From(c).WithError(err).Error("Invalid authentication params.")
```
Comment (FilipNikolovski) [inline actions/auth.go:51]: No need to log validation.

File: entities/params/postsignup.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package params
+
+// PostSignUp represents request body for POST /api/signup
+type PostSignUp struct {
+	Email         string `form:"email" validate:"required,email"`
+	Password      string `form:"password" validate:"required,min=8"`
+	TokenResponse string `form:"token_response" validate:"optional"`
+}
+
+func (p *PostSignUp) TrimSpaces() {
```
Comment (FilipNikolovski) [inline entities/params/postsignup.go:10]: It's usually okay to add a comment here inside the method explaining that we don't need to do any opertion, like:

```go
func (p *PostSignUp) TrimSpaces() {
// no-op
}
```

File: actions/auth.go (hunk 1 above)
Comment (FilipNikolovski) [reply actions/auth.go:43]: What is the error here in case ShouldBind fails?

**Review by FilipNikolovski — CHANGES_REQUESTED** [review]

File: actions/auth.go (hunk 1 above)
Comment (FilipNikolovski) [reply actions/auth.go:43]: After discussion, we should remove the logs from ShouldBind and Validate. Resolving this.

File: entities/params/auth.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+	Password string `form:"password" validate:"required"`
+}
+
+func (p *PostAuthenticate) TrimSpaces() {
+}
+
+// PostSignUp represents request body for POST /api/signup
+type PostSignUp struct {
+	Email         string `form:"email" validate:"required,email"`
+	Password      string `form:"password" validate:"required,min=8,max=191"`
```
Comment (FilipNikolovski) [inline entities/params/auth.go:15]: @Gligor23 we don't limit passwords, remove the max=191 validation from all Password fields.

**Review by Dzalevski — APPROVED** [review]

File: entities/params/auth.go [hunk 6]
```diff
@@ -1,21 +1,25 @@
...
+import "strings"
+
 // PostAuthenticate represents request body for POST /api/authenticate
 type PostAuthenticate struct {
 	Username string `form:"username" validate:"required,email"`
 	Password string `form:"password" validate:"required"`
 }
 
 func (p *PostAuthenticate) TrimSpaces() {
+	p.Password = strings.TrimSpace(p.Password)
```
Comment (FilipNikolovski) [inline entities/params/auth.go:12]: Don't trim passwords, we treat passwords as the user entered them.

File: entities/params/auth.go [hunk 7]
```diff
@@ -1,21 +1,25 @@
...
 // PostSignUp represents request body for POST /api/signup
 type PostSignUp struct {
 	Email         string `form:"email" validate:"required,email"`
-	Password      string `form:"password" validate:"required,min=8,max=191"`
+	Password      string `form:"password" validate:"required,min=8"`
 	TokenResponse string `form:"token_response" validate:"optional"`
 }
 
 func (p *PostSignUp) TrimSpaces() {
+	p.Password = strings.TrimSpace(p.Password)
```
Comment (FilipNikolovski) [inline entities/params/auth.go:23]: Here as well.

**Review by FilipNikolovski — CHANGES_REQUESTED** [review]

**Review by FilipNikolovski — APPROVED** [review]

**Pull request merged**